src/framework/auth/auth.module.ts

```diff
@@ -33,13 +41,24 @@ import { routes } from './auth.routes';
 import { deepExtend } from './helpers';
 
 export function nbStrategiesFactory(options: NbAuthOptions, injector: Injector): NbAuthStrategy[] {
-  return options.strategies
-    .map(([strategyClass, strategyOptions]: [NbAuthStrategyClass, NbAuthStrategyOptions]) => {
+  const strategies = [];
+  options.strategies
+    .forEach(([strategyClass, strategyOptions]: [NbAuthStrategyClass, NbAuthStrategyOptions]) => {
       const strategy: NbAuthStrategy = injector.get(strategyClass);
       strategy.setOptions(strategyOptions);
 
-      return strategy;
+      strategies.push(strategy);
+    });
+  return strategies;
+}
+
+export function nbTokensFactory(strategies: NbAuthStrategy[]): NbAuthTokenClass[] {
```

Comment:
Also, It would be great if you add tokens validation. We have to warn a user about missed and same tokens.

Reply:
actutlly it could be the case, when two strategies use the same token with same `NAME`